packages/arb-bridge-eth/contracts/validator/ValidatorUtils.sol

```diff
@@ -265,4 +265,14 @@ contract ValidatorUtils {
         }
         return (challenges, hasMoreStakers);
     }
+
+    function areUnresolvedNodesLinear(Rollup rollup) external view returns (bool) {
```

Comment:
Might be good to add some documentation here that it's O(n) in the number of nodes, but safe since in the happy case where there are no forks we want assume that there won't be too many nodes and in the malicious case the function will terminate early

Reply:
I think a better way to describe it is O(depth) but yeah I'll document that